webapp/lib/controller.dart

```diff
@@ -881,47 +881,64 @@ void updateViewForConversation(model.Conversation conversation) {
   }
 }
 
-void sendReply(model.SuggestedReply reply, model.Conversation conversation) {
+void sendReply(model.SuggestedReply reply, model.Conversation conversation) async {
   model.Message newMessage = new model.Message()
     ..text = reply.text
     ..datetime = new DateTime.now()
     ..direction = model.MessageDirection.Out
     ..translation = reply.translation
     ..tagIds = [];
   conversation.messages.add(newMessage);
-  view.conversationPanelView.addMessage(
-    new view.MessageView(
+  var newMessageView = new view.MessageView(
       newMessage.text,
       newMessage.datetime,
       conversation.docId,
       conversation.messages.indexOf(newMessage),
       translation: newMessage.translation,
-      incoming: false)
-  );
-  platform.sendMessage(conversation.docId, reply.text);
+      incoming: false);
+  view.conversationPanelView.addMessage(newMessageView);
+  try {
+    await platform.sendMessage(conversation.docId, reply.text);
```

Comment:
What are your thoughts to adding an (optional) `onError` parameter to `sendMessage`? I could be persuaded both ways, but I'm wondering if it would be nicer to catch the `PubSubException` at platform level, rather than here?

Reply:
@marianamarasoiu Good suggestion. I think it looks better with `onError` as you suggest
and gives us a central place to make error handling changes... https://github.com/larksystems/nook/pull/331